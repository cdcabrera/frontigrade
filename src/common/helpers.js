import _ from 'lodash';
import moment from 'moment/moment';

const copyClipboard = text => {
  let successful;

  try {
    window.getSelection().removeAllRanges();

    const newTextarea = document.createElement('pre');
    newTextarea.appendChild(document.createTextNode(text));

    newTextarea.style.position = 'absolute';
    newTextarea.style.top = '-1000px';
    newTextarea.style.left = '-1000px';
    newTextarea.style.overflow = 'hidden';
    newTextarea.style.width = '1px';
    newTextarea.style.height = '1px';

    const range = document.createRange();
    window.document.body.appendChild(newTextarea);

    range.selectNode(newTextarea);

    window.getSelection().addRange(range);

    successful = window.document.execCommand('copy');

    window.document.body.removeChild(newTextarea);
    window.getSelection().removeAllRanges();
  } catch (e) {
    successful = false;
    console.warn('Copy to clipboard failed.', e.message);
  }

  return successful;
};

const generateId = prefix => `${prefix || 'generatedid'}-${Math.ceil(1e5 * Math.random())}`;

const generateHoursFromSeconds = seconds => {
  let parsedSeconds = Number.parseFloat(seconds);
  parsedSeconds = Number.isNaN(parsedSeconds) ? null : parsedSeconds;

  parsedSeconds =
    parsedSeconds === null ? parsedSeconds : Math.floor(moment.duration(parsedSeconds, 'seconds').asHours());

  return parsedSeconds;
};

const generatePriorYearMonthArray = () => {
  const dateStart = moment().subtract(12, 'months');
  const dateEnd = moment().startOf('month');

  let timeValues = [];

  while (dateEnd > dateStart || dateStart.format('M') === dateEnd.format('M')) {
    timeValues.push({
      title: dateStart.startOf('month').format('YYYY MMMM'),
      start: dateStart.startOf('month').toISOString(),
      end: dateStart.endOf('month').toISOString()
    });

    dateStart.add(1, 'month');
  }

  timeValues = timeValues.reverse();
  timeValues[0] = {
    default: true,
    title: 'Last 30 Days',
    start: moment()
      .startOf('day')
      .subtract(30, 'days')
      .toISOString(),
    end: moment()
      .endOf('day')
      .toISOString()
  };

  return { defaultTime: timeValues[0], timeValues };
};

const getMessageFromResults = (results, filterField = null) => {
  const status = _.get(results, 'response.status', results.status);
  const statusResponse = _.get(results, 'response.statusText', results.statusText);
  const messageResponse = _.get(results, 'response.data', results.message);
  const detailResponse = _.get(results, 'response.data', results.detail);

  if (status < 400 && !messageResponse && !detailResponse) {
    return statusResponse;
  }

  if (status >= 500 || (status === undefined && (messageResponse || detailResponse))) {
    return `${status || ''} Server is currently unable to handle this request. ${messageResponse ||
      detailResponse ||
      ''}`;
  }

  if (typeof messageResponse === 'string') {
    return messageResponse;
  }

  if (typeof detailResponse === 'string') {
    return detailResponse;
  }

  const getMessages = (messageObject, filterKey) => {
    const obj = filterKey ? messageObject[filterKey] : messageObject;

    return _.map(
      obj,
      next => {
        if (_.isArray(next)) {
          return getMessages(next);
        }

        return next;
      },
      null
    );
  };

  return _.join(getMessages(messageResponse || detailResponse, filterField), '\n');
};

const getStatusFromResults = results => {
  let status = _.get(results, 'response.status', results.status);

  if (status === undefined) {
    status = 0;
  }

  return status;
};

const noop = Function.prototype;

const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state };

  if (prop && !state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (reset && prop && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset && prop) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else if (reset && !prop) {
    obj = {
      ...state,
      ...initialState,
      ...data
    };
  } else if (prop) {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  } else {
    obj = {
      ...state,
      ...data
    };
  }

  return obj;
};

const prettyPrintJson = json => JSON.stringify(json, null, 2);

const DEV_MODE = process.env.REACT_APP_ENV === 'development';

const OC_MODE = process.env.REACT_APP_ENV === 'oc';

const RH_BRAND = process.env.REACT_APP_RH_BRAND === 'true';

const UI_COMMIT_HASH = process.env.REACT_APP_UI_COMMIT_HASH;

const FULFILLED_ACTION = (base = '') => `${base}_FULFILLED`;

const PENDING_ACTION = (base = '') => `${base}_PENDING`;

const REJECTED_ACTION = (base = '') => `${base}_REJECTED`;

const HTTP_STATUS_RANGE = status => `${status}_STATUS_RANGE`;

const helpers = {
  copyClipboard,
  generateId,
  generateHoursFromSeconds,
  generatePriorYearMonthArray,
  getMessageFromResults,
  getStatusFromResults,
  noop,
  setStateProp,
  prettyPrintJson,
  DEV_MODE,
  OC_MODE,
  RH_BRAND,
  UI_COMMIT_HASH,
  FULFILLED_ACTION,
  PENDING_ACTION,
  REJECTED_ACTION,
  HTTP_STATUS_RANGE
};

export { helpers as default, helpers };
