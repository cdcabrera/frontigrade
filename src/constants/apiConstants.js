const API_RESPONSE_ACCOUNTS = 'cloud_account_overviews';
const API_RESPONSE_ACCOUNTS_ID = 'id';
const API_RESPONSE_ACCOUNTS_IMAGES = 'images';
const API_RESPONSE_ACCOUNTS_INSTANCES = 'instances';
const API_RESPONSE_ACCOUNTS_NAME = 'name';
const API_RESPONSE_ACCOUNTS_OPENSHIFT = 'openshift_instances';
const API_RESPONSE_ACCOUNTS_RHEL = 'rhel_instances';
const API_RESPONSE_ACCOUNTS_START = 'start';
const API_RESPONSE_ACCOUNTS_END = 'end';
const API_RESPONSE_ACCOUNTS_DATE = 'creation_date';
const API_QUERY_START = 'start';
const API_QUERY_END = 'end';
const API_QUERY_NAME = 'name_pattern';

// ToDo: need to go back filter these down, currently have repeats
const API_GET_ACCOUNT_COUNT = 'count';
const API_GET_ACCOUNT_RESULTS = 'results';
const API_ACCOUNT_ARN = 'account_arn';
const API_ACCOUNT_NAME = 'name';
const API_RESOURCE_TYPE = 'resourcetype';
const API_AWS_ACCOUNT_ID = 'aws_account_id';

const apiTypes = {
  API_RESPONSE_ACCOUNTS,
  API_RESPONSE_ACCOUNTS_ID,
  API_RESPONSE_ACCOUNTS_IMAGES,
  API_RESPONSE_ACCOUNTS_INSTANCES,
  API_RESPONSE_ACCOUNTS_NAME,
  API_RESPONSE_ACCOUNTS_OPENSHIFT,
  API_RESPONSE_ACCOUNTS_RHEL,
  API_RESPONSE_ACCOUNTS_START,
  API_RESPONSE_ACCOUNTS_END,
  API_RESPONSE_ACCOUNTS_DATE,
  API_QUERY_START,
  API_QUERY_END,
  API_QUERY_NAME,
  API_GET_ACCOUNT_COUNT,
  API_GET_ACCOUNT_RESULTS,
  API_ACCOUNT_ARN,
  API_ACCOUNT_NAME,
  API_RESOURCE_TYPE,
  API_AWS_ACCOUNT_ID
};

export {
  apiTypes as default,
  apiTypes,
  API_RESPONSE_ACCOUNTS,
  API_RESPONSE_ACCOUNTS_ID,
  API_RESPONSE_ACCOUNTS_IMAGES,
  API_RESPONSE_ACCOUNTS_INSTANCES,
  API_RESPONSE_ACCOUNTS_NAME,
  API_RESPONSE_ACCOUNTS_OPENSHIFT,
  API_RESPONSE_ACCOUNTS_RHEL,
  API_RESPONSE_ACCOUNTS_START,
  API_RESPONSE_ACCOUNTS_END,
  API_RESPONSE_ACCOUNTS_DATE,
  API_QUERY_START,
  API_QUERY_END,
  API_QUERY_NAME,
  API_GET_ACCOUNT_COUNT,
  API_GET_ACCOUNT_RESULTS,
  API_ACCOUNT_ARN,
  API_ACCOUNT_NAME,
  API_RESOURCE_TYPE,
  API_AWS_ACCOUNT_ID
};
