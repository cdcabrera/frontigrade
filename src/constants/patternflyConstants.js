import { BackgroundImageSrc } from '@patternfly/react-core';
import bgFilter from '@patternfly/patternfly-next/assets/images/background-filter.svg';
import pfBg576 from '@patternfly/patternfly-next/assets/images/pfbg_576.jpg';
import pfBg576R2x from '@patternfly/patternfly-next/assets/images/pfbg_576@2x.jpg';
import pfBg768 from '@patternfly/patternfly-next/assets/images/pfbg_768.jpg';
import pfBg768R2x from '@patternfly/patternfly-next/assets/images/pfbg_768@2x.jpg';
import pfBg992 from '@patternfly/patternfly-next/assets/images/pfbg_992.jpg';
import pfBg992R2x from '@patternfly/patternfly-next/assets/images/pfbg_992@2x.jpg';
import pfBg1200 from '@patternfly/patternfly-next/assets/images/pfbg_1200.jpg';
import pfBg2000 from '@patternfly/patternfly-next/assets/images/pfbg_2000.jpg';

export const bgImages = {
  [BackgroundImageSrc.lg]: pfBg1200,
  [BackgroundImageSrc.md]: pfBg992,
  [BackgroundImageSrc.md2x]: pfBg992R2x,
  [BackgroundImageSrc.sm]: pfBg768,
  [BackgroundImageSrc.sm2x]: pfBg768R2x,
  [BackgroundImageSrc.xl]: pfBg2000,
  [BackgroundImageSrc.xs]: pfBg576,
  [BackgroundImageSrc.xs2x]: pfBg576R2x,
  [BackgroundImageSrc.filter]: `${bgFilter}#image_overlay`
};

export default bgImages;
