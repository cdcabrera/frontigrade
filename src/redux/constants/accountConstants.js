const ADD_ACCOUNT = 'ADD_ACCOUNT';
const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
const DELETE_ACCOUNTS = 'DELETES_ACCOUNTS';
const GET_ACCOUNT = 'GET_ACCOUNT';
const GET_ACCOUNTS = 'GET_ACCOUNTS';
const UPDATE_ACCOUNTS = 'UPDATE_ACCOUNTS';
const UPDATE_ACCOUNT = 'UPDATE_ACCOUNT';
const UPDATE_ACCOUNT_FIELD = 'UPDATE_ACCOUNT_FIELD';
const UPDATE_ACCOUNT_IMAGE = 'UPDATE_ACCOUNT_IMAGE';
const UPDATE_ACCOUNT_IMAGE_FIELD = 'UPDATE_ACCOUNT_IMAGE_FIELD';
const UPDATE_ACCOUNT_IMAGES_INSTANCES = 'UPDATE_ACCOUNT_IMAGES_INSTANCES';
const GET_ACCOUNT_IMAGES = 'GET_ACCOUNT_IMAGES';
const GET_ACCOUNT_IMAGES_INSTANCES = 'GET_ACCOUNT_IMAGES_INSTANCES';
const GET_ACCOUNT_INSTANCES = 'GET_ACCOUNT_INSTANCES';
// ToDo: clean up account constants, ADD_ACCOUNT_WIZARD not used
const ADD_ACCOUNT_WIZARD = 'ADD_ACCOUNT_WIZARD';
const ADD_ACCOUNT_WIZARD_STEP_POLICY = 'ADD_ACCOUNT_WIZARD_STEP_POLICY';
const INVALID_ACCOUNT_WIZARD_STEP_POLICY = 'INVALID_ACCOUNT_WIZARD_STEP_POLICY';
const ADD_ACCOUNT_WIZARD_STEP_ROLE = 'ADD_ACCOUNT_WIZARD_STEP_ROLE';
const INVALID_ACCOUNT_WIZARD_STEP_ROLE = 'INVALID_ACCOUNT_WIZARD_STEP_ROLE';
const ADD_ACCOUNT_WIZARD_STEP_ARN = 'ADD_ACCOUNT_WIZARD_STEP_ARN';
const INVALID_ACCOUNT_WIZARD_STEP_ARN = 'INVALID_ACCOUNT_WIZARD_STEP_ARN';
// ToDo: end clean up account constants, poorly named, aim at using ADD_ACCOUNT_WIZARD_ prefix
const ADD_ACCOUNT_SHOW = 'ADD_ACCOUNT_SHOW';
const UPDATE_ACCOUNT_HIDE = 'UPDATE_ACCOUNT_HIDE';

const DELETE_ACCOUNT_HIDE = 'DELETE_ACCOUNT_HIDE';
const DELETE_ACCOUNT_SHOW = 'DELETE_ACCOUNT_SHOW';
const EDIT_ACCOUNT_HIDE = 'EDIT_ACCOUNT_HIDE';
const EDIT_ACCOUNT_SHOW = 'EDIT_ACCOUNT_SHOW';

export {
  ADD_ACCOUNT,
  DELETE_ACCOUNT,
  DELETE_ACCOUNTS,
  GET_ACCOUNT,
  GET_ACCOUNTS,
  UPDATE_ACCOUNTS,
  UPDATE_ACCOUNT,
  UPDATE_ACCOUNT_FIELD,
  UPDATE_ACCOUNT_IMAGE,
  UPDATE_ACCOUNT_IMAGE_FIELD,
  UPDATE_ACCOUNT_IMAGES_INSTANCES,
  GET_ACCOUNT_IMAGES,
  GET_ACCOUNT_IMAGES_INSTANCES,
  GET_ACCOUNT_INSTANCES,
  ADD_ACCOUNT_WIZARD,
  ADD_ACCOUNT_WIZARD_STEP_POLICY,
  INVALID_ACCOUNT_WIZARD_STEP_POLICY,
  ADD_ACCOUNT_WIZARD_STEP_ROLE,
  INVALID_ACCOUNT_WIZARD_STEP_ROLE,
  ADD_ACCOUNT_WIZARD_STEP_ARN,
  INVALID_ACCOUNT_WIZARD_STEP_ARN,
  ADD_ACCOUNT_SHOW,
  UPDATE_ACCOUNT_HIDE,
  DELETE_ACCOUNT_HIDE,
  DELETE_ACCOUNT_SHOW,
  EDIT_ACCOUNT_HIDE,
  EDIT_ACCOUNT_SHOW
};
