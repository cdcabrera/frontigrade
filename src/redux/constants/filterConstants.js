const TOOLBAR_CLEAR_FILTERS = 'TOOLBAR_CLEAR_FILTERS';
const TOOLBAR_ADD_FILTER = 'TOOLBAR_ADD_FILTER';
const TOOLBAR_REMOVE_FILTER = 'TOOLBAR_REMOVE_FILTER';
const TOOLBAR_SET_DATE_TYPE = 'TOOLBAR_SET_DATE_TYPE';
const TOOLBAR_CLEAR_DATE_TYPE = 'TOOLBAR_CLEAR_DATE_TYPE';
const TOOLBAR_SET_FILTER_TYPE = 'TOOLBAR_SET_FILTER_TYPE';
const TOOLBAR_SET_FILTER_VALUE = 'TOOLBAR_SET_FILTER_VALUE';
const TOOLBAR_SET_SORT_TYPE = 'TOOLBAR_SET_SORT_TYPE';
const TOOLBAR_TOGGLE_SORT_ASCENDING = 'TOOLBAR_TOGGLE_SORT_ASCENDING';

export {
  TOOLBAR_ADD_FILTER,
  TOOLBAR_CLEAR_FILTERS,
  TOOLBAR_REMOVE_FILTER,
  TOOLBAR_SET_DATE_TYPE,
  TOOLBAR_CLEAR_DATE_TYPE,
  TOOLBAR_SET_FILTER_TYPE,
  TOOLBAR_SET_FILTER_VALUE,
  TOOLBAR_SET_SORT_TYPE,
  TOOLBAR_TOGGLE_SORT_ASCENDING
};
