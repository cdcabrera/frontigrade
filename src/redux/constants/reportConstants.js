const REPORT_EXPORT = 'REPORT_EXPORT';

export { REPORT_EXPORT as default, REPORT_EXPORT };
