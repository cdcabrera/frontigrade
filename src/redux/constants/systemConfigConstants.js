const GET_SYSTEM_CONFIG = 'GET_SYSTEM_CONFIG';

export { GET_SYSTEM_CONFIG as default, GET_SYSTEM_CONFIG };
