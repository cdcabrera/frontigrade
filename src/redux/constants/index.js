import * as accountTypes from './accountConstants';
import * as confirmationModalTypes from './confirmationModalConstants';
import * as systemConfigTypes from './systemConfigConstants';
import * as toastNotificationTypes from './toastNotificationConstants';
import * as userTypes from './userConstants';

const reduxTypes = {
  account: accountTypes,
  confirmationModal: confirmationModalTypes,
  systemConfig: systemConfigTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  accountTypes,
  confirmationModalTypes,
  systemConfigTypes,
  toastNotificationTypes,
  userTypes
};
