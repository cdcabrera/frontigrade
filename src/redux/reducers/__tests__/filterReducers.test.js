import { filterReducers } from '../';
import { filterTypes as types, accountTypes } from '../../constants';
import helpers from '../../../common/helpers';

describe('FilterReducers', () => {
  it('should return the initial state', () => {
    expect(filterReducers.initialState).toBeDefined();
  });

  it('should handle all defined types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = filterReducers(undefined, dispatched);

      /**
       * ToDo: API-no-default-list
       * API currently has no default values, we set these defaults in state as temp-fix.
       * This condition should be removed.
       */
      if (resultState.account && resultState.account.query) {
        delete resultState.account.query;
      }

      expect({ type: value, result: resultState }).toMatchSnapshot('defined types');
    });
  });

  it('should handle specific account fulfilled types', () => {
    const specificTypes = [accountTypes.GET_ACCOUNTS];

    specificTypes.forEach(value => {
      if (/wizard/i.test(value)) {
        return;
      }

      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            count: 1,
            results: ['success']
          }
        }
      };

      const resultState = filterReducers(undefined, dispatched);

      /**
       * ToDo: API-no-default-list
       * API currently has no default values, we set these defaults in state as temp-fix.
       * This condition should be removed.
       */
      if (resultState.account && resultState.account.query) {
        delete resultState.account.query;
      }

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot('fulfilled types');
    });
  });
});
