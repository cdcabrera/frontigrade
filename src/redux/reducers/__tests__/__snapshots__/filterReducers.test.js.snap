// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined types 1`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_ADD_FILTER",
}
`;

exports[`defined types 2`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_CLEAR_FILTERS",
}
`;

exports[`defined types 3`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_REMOVE_FILTER",
}
`;

exports[`defined types 4`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_SET_DATE_TYPE",
}
`;

exports[`defined types 5`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_SET_FILTER_TYPE",
}
`;

exports[`defined types 6`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_SET_FILTER_VALUE",
}
`;

exports[`defined types 7`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_SET_SORT_TYPE",
}
`;

exports[`defined types 8`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "TOOLBAR_TOGGLE_SORT_ASCENDING",
}
`;

exports[`fulfilled types 1`] = `
Object {
  "result": Object {
    "account": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "dateValue": null,
      "expandedItems": Array [],
      "filterField": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedCount": 0,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortValue": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "detail": Object {},
    "images": Object {},
  },
  "type": "GET_ACCOUNTS_FULFILLED",
}
`;
